### PR TITLE
Update habitica documentation

### PR DESCRIPTION
**Pull request by tr4nt0r**
## Proposed change

Final changes for Habitica documentation, update descriptions for sensor entities, add descriptions for newly added button and switch entities and for the new to-do entities. References to deprecated entities have been removed.

- https://github.com/home-assistant/core/pull/121472
- https://github.com/home-assistant/core/pull/121461
- https://github.com/home-assistant/core/pull/116655

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/121472
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Enhanced and clarified documentation for the Habitica integration within Home Assistant.
	- Restructured overview emphasizing progress monitoring and task management.
	- Expanded player data section with a comprehensive and readable list of available sensors.
	- Introduced new sections detailing button and switch controls for user actions, with clear examples and effects.

### CODE REVIEW

File: source/_integrations/habitica.markdown [hunk 1]
```diff
@@ -20,40 +20,42 @@ ha_config_flow: true
...
-- Player's health points
-- Player's max health
-- Player's mana points
-- Player's max mana points
-- Player's experience
-- Player's experience to the next level
-- Player's level
-- Player's gold pieces
-- Player's class
+- **Class:** Indicates the class of your character (Warrior, Rogue, Healer or Mage).
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
- **Class:** Indicates the class of your character (Warrior, Rogue, Healer, or Mage).
```

File: source/_integrations/habitica.markdown [hunk 2]
```diff
@@ -20,40 +20,42 @@ ha_config_flow: true
...
 The following Habitica tasks are available as to-do lists in Home Assistant. You can add, delete, edit and check-off completed tasks
 
-- To-do's
-- Dailies
+- **To-Do's:** Displays a comprehensive list of active and completed to-dos. Each to-do includes its due date if applicable, allowing you to check them off, edit them, delete them, and create new to-dos seamlessly.
+- **Dailies:** Shows the daily tasks that need to be completed today or in the future. Tasks completed yesterday can still be marked off as "yesterdailies" until a new day starts.
+
+## Button controls
+
+- **Start my day:** Initiates daily routine actions in Habitica, including resetting your dailies, deal damage from unfinished dailies and quest bosses, habits adjustment, buff expiration and mana regeneration based on completed dailies.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
- **Start my day:** Initiates daily routine actions in Habitica, including resetting your dailies, deal damage from unfinished dailies and quest bosses, habits adjustment, buff expiration, and mana regeneration based on completed dailies.
```

File: source/_integrations/habitica.markdown [hunk 3]
```diff
@@ -142,12 +144,12 @@ Also an event `habitica_api_call_success` will be fired with the following data:
 
 ## Templating
 
-`sensor.habitica_USER_dailys`, `sensor.habitica_USER_habits`, `sensor.habitica_USER_rewards`, and `sensor.habitica_USER_todos` have state attributes listing the user's respective tasks. For example, you can see this information in **Developer Tools** -> **States** -> `sensor.habitica_USER_dailys` -> **Attributes**, or by adding a [Markdown card](/dashboards/markdown/) to a dashboard with the following code:
+`sensor.habitica_USER_habits` and `sensor.habitica_USER_rewards` have state attributes listing the user's respective tasks. For example, you can see this information in **Developer Tools** -> **States** -> `sensor.habitica_USER_habits` -> **Attributes**, or by adding a [Markdown card](/dashboards/markdown/) to a dashboard with the following code:
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
`sensor.habitica_USER_habits` and `sensor.habitica_USER_rewards` have state attributes listing the user's respective tasks. For example, you can see this information in **{% my developer_states title="Developer Tools > States" %}** under `sensor.habitica_USER_habits` > **Attributes**, or by adding a [Markdown card](/dashboards/markdown/) to a dashboard with the following code:
```

File: source/_integrations/habitica.markdown [hunk 4]
```diff
@@ -20,40 +20,42 @@ ha_config_flow: true
 ha_integration_type: integration
 ---
 
-This integration allows you to monitor and manage your Habitica profile. This integration exposes the [Habitica's API](https://habitica.com/apidoc/) as a Home Assistant service. It supports multiple users and allows you to automate checking out your habits and daily tasks or casting magics using Home Assistant.
+The Habitca {% term integration %} enables you to monitor your adventurer's progress and stats in Home Assistant and seamlessly integrates your to-do's and daily tasks.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown:23]: Are there any Prerequisites? Do I need an account, user name, API key etc. before configuring the integration in Home Assistant?

Comment (tr4nt0r) [reply source/_integrations/habitica.markdown:23]: There haven't been any changes to the setup process but it is described in the config flow how to authenticate with API key and token. Currently waiting for a PR to be approved that will further streamline the setup process of the integration:
- https://github.com/home-assistant/core/pull/117622

I'd suggest to add a section about setup and prerequisites to the documentation after that one was merged.

Comment (c0ffeeca7) [reply source/_integrations/habitica.markdown:23]: I agree, it is outside the scope of this PR. Maybe together with 117622?

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @tr4nt0r :+1:

**Pull request merged**